packages/tracing/test/browser/request.test.ts

```diff
@@ -84,106 +91,149 @@ describe('_fetchCallback()', () => {
-      url: '/users',
+        type: 'fetch',
+        url: 'http://dogs.are.great/',
+      });
+      expect(newSpan!.description).toBe('GET http://dogs.are.great/');
+      expect(newSpan!.op).toBe('http');
+      expect(fetchHandlerData.fetchData?.__span).toBeDefined();
+
+      const postRequestFetchHandlerData = {
+        ...fetchHandlerData,
+        endTimestamp,
+      };
+
+      // triggered by response coming back
+      fetchCallback(postRequestFetchHandlerData, alwaysCreateSpan, spans);
+
+      expect(newSpan!.endTimestamp).toBeDefined();
+    });
+
+    it('sets response status on finish', () => {
+      const spans: Record<string, Span> = {};
+
+      // triggered by request being sent
+      fetchCallback(fetchHandlerData, alwaysCreateSpan, spans);
+
+      const newSpan = transaction.spanRecorder?.spans[1];
+
+      expect(newSpan).toBeDefined();
+
+      const postRequestFetchHandlerData = {
+        ...fetchHandlerData,
+        endTimestamp,
+        response: { status: 404 } as Response,
+      };
+
+      // triggered by response coming back
+      fetchCallback(postRequestFetchHandlerData, alwaysCreateSpan, spans);
+
+      expect(newSpan!.status).toBe(SpanStatus.fromHttpCode(404));
+    });
+
+    it('adds sentry-trace header to fetch requests', () => {
+      // TODO
     });
-    expect(fetchSpan.description).toBe('GET /users');
-    expect(fetchSpan.op).toBe('http');
-    if (data.fetchData) {
-      expect(data.fetchData.__span).toBeDefined();
-    } else {
-      fail('Fetch data does not exist');
-    }
-
-    const newData = {
-      ...data,
-      endTimestamp: data.startTimestamp + 12343234,
-    };
-
-    // End fetch request
-    _fetchCallback(newData, shouldCreateSpan, spans);
-    expect(spans).toEqual({});
-    if (transaction.spanRecorder) {
-      expect(transaction.spanRecorder.spans[1].endTimestamp).toBeDefined();
-    } else {
-      fail('Transaction does not have span recorder');
-    }
   });
 
-  it('sets response status on finish', () => {
-    const shouldCreateSpan = (_: string): boolean => true;
-    const data: FetchData = {
-      args: ['/users'],
-      fetchData: {
+  describe('xhrCallback()', () => {
+    it('does not create span if shouldCreateSpan returns false', () => {
+      const spans = {};
+
+      xhrCallback(xhrHandlerData, neverCreateSpan, spans);
+
+      expect(spans).toEqual({});
+    });
+
+    it('adds sentry-trace header to XHR requests', () => {
+      xhrCallback(xhrHandlerData, alwaysCreateSpan, {});
+
+      expect(setRequestHeader).toHaveBeenCalledWith(
+        'sentry-trace',
+        expect.stringMatching(tracingUtils.TRACEPARENT_REGEXP),
+      );
+    });
+
+    it('creates and finishes XHR span on active transaction', () => {
+      const spans = {};
+
+      // triggered by request being sent
+      xhrCallback(xhrHandlerData, alwaysCreateSpan, spans);
+
+      const newSpan = transaction.spanRecorder?.spans[1];
+
+      expect(newSpan).toBeDefined();
```

Comment:
Writing such expectations is kinda redundant, because `toBeInstanceOf` already verifies that span is defined in the first place.